nc-games/src/Components/NewComment.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react";
+import { postComment } from "../api";
+
+function NewComment({reviewId, hasPostedComment, setHasPostedComment, setNumComments, setComments}) {
+    const [newComment, setNewComment] = useState("");
+    const [isPostingError, setIsPostingError] = useState(false)
+    const [hasPosted, setHasPosted] = useState(false)
+    const [hasPostedError, setHasPostedError] = useState(false)
+    const [zeroLength, setZeroLength] = useState(false)
+    const [user, setUser] = useState("grumpy19")
+
+    function handleSubmit(event) {
+        event.preventDefault();
+        setZeroLength(false)
+        setHasPostedComment(false);
+        setHasPostedError(false);
+        if(!hasPosted && newComment.length !== 0) {
+            setComments(curr => [newComment, ...curr])
+            setNumComments((curr) => curr + 1)
+            setHasPosted(true);
+            postComment(user, reviewId, newComment).then(response => {
+                setHasPosted(false)
+                setHasPostedComment(true)
+                setNewComment("")
+                event.target.reset();
+            }).catch(e => {
+                setComments(curr => curr.slice(-(curr.length - 1)))
+                setIsPostingError(true)
+                setHasPosted(false)
+                setNumComments((curr) => curr - 1)
+            })
+        } else {
+            if(newComment.length === 0) {
+                setZeroLength(true)
+            } else {
+                setHasPostedError(true);
+                setHasPosted(false)
+            }
```

Comment:
This could benefit from a comment, as i've no idea what it does or why its needed. 
Would be hard for someone else to take over this code. They wouldnt change this, as they'd presume it must be there for a reason